File: iter1/src/pagePackage/LoginPage.java
package pagePackage;

import systemMessagePackage.*;
import userInfoPackage.UserInfo;

public class LoginPage extends Page{
	private UserInfo userInfo;

	public LoginPage(String content) {
		super(content);
		setType(PageType.LOGIN_PAGE);
		setName("Login Page");
		userInfo = new UserInfo(null, null);
	}


	@Override
	public SystemMessage runPage() {
		// set user info null because this func may not run one time
		userInfo.reset();
		
		// print the login page content to console
		showContent();
		
		// take password
		System.out.println("Username:");
		userInfo.setUsername(takeInput());
		
		// take password
		System.out.println("Password:");
		userInfo.setPassword(takeInput());
			
		return new SystemMessage(FunctionType.LOGIN, null,userInfo);
	}

	
	//////// GETTER - SETTET METHODS ///////////
	public UserInfo getUserInfo() {
		return userInfo;
	}

	public void setUserInfo(UserInfo userInfo) {
		this.userInfo = userInfo;
	}	
}

File: iter3/python_development/Page/LoginPage.py
from iter3.python_development.Page.Page import Page
from iter3.python_development.Page.PageType import PageType
from iter3.python_development.System.FunctionType import FunctionType
from iter3.python_development.System.SystemMessage import SystemMessage


class LoginPage(Page):
    
    def __init__(self, content):
        super().__init__(content)
        self.setType("LOGIN_PAGE")
        self.setName("Login Page")
    
    def runPage(self):
        self.showContent()
        userInfo = [None, None]
        userInfo[0] = input("Username: ")
        userInfo[1] = input("Password: ")

        return SystemMessage("LOGIN", None,userInfo)